### PR TITLE
Static ip check

### PR DESCRIPTION
**Pull request by sgaoshang**
Add vSphere static IP e2e check

TC: [OCP-66218](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-66218)

### CODE REVIEW

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
@jinyunma pls help to review it, thanks.

File: ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+set -o pipefail
+
+STATIC_IPS="${SHARED_DIR}"/static-ip-hosts.txt
+
+static_ips=$(yq-go r "$STATIC_IPS" 'hosts.*.networkDevice.ipAddrs.*' | cut -d "/" -f 1)
+node_ips=$(oc get nodes -o=jsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+
+for node_ip in $node_ips; do
+    # shellcheck disable=SC2199
+    if [[ ${static_ips[@]} =~ $node_ip ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh:14]:
```suggestion
    if [[ " ${static_ips[@]} " =~ " $node_ip " ]]; then
```

File: ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+set -o nounset
+set -o errexit
+set -o pipefail
+
+STATIC_IPS="${SHARED_DIR}"/static-ip-hosts.txt
+
+static_ips=$(yq-go r "$STATIC_IPS" 'hosts.*.networkDevice.ipAddrs.*' | cut -d "/" -f 1)
+node_ips=$(oc get nodes -o=jsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+
+for node_ip in $node_ips; do
```
Comment (jinyunma) [inline ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh:12]: If ip is not assigned correctly, from output message, seems unable to know on which node.
How about to loop each node name, then get node ip and check if it is in static_ips[@]?

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-static-ip-tp-f28

**Comment by sgaoshang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-ipi-static-ip-tp-f28

File: ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+set -o errexit
+set -o pipefail
+
+check_result=0
+
+STATIC_IPS="${SHARED_DIR}"/static-ip-hosts.txt
+static_ips=$(yq-go r "$STATIC_IPS" 'hosts.*.networkDevice.ipAddrs.*' | awk -F "/" '{print $1}')
+nodes_list=$(oc get nodes -ojson | jq -r '.items[].metadata.name')
+
+for node in $nodes_list; do
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh:13]:
```suggestion
node_check_result=0
for node in $nodes_list; do
```

File: ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    node_check_result=1
+    node_ip=$(oc get nodes "$node" -o=jsonpath='{.status.addresses[?(@.type=="InternalIP")].address}')
+    for static_ip in $static_ips; do
+        if [ "$static_ip" == "$node_ip" ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/vsphere/staticip/cucushift-installer-check-vsphere-staticip-commands.sh:17]:
```suggestion
        if [[ "$static_ip" == "$node_ip" ]]; then
```

**Comment by sgaoshang** [comment]
cc: @WenXinWei

**Pull request merged**